app/code/Magento/Sales/Block/Adminhtml/Order/Create/Form.php

```diff
@@ -182,7 +182,7 @@ public function getOrderDataJson()
                 );
             }
         }
-        if ($this->getStoreId() !== null) {
```

Comment:
I'm not sure we really need to remove this strict condition?
Could you please double check if it was not removed accidentally?

Reply:
I have checked in single store it selects frontend default store and in multi store it asks to select frontend store then quote is created with selected store id.
So I thought to keep the strict condition.

Reply:
I have reverted to restrict condition in latest commit, please verify.